### PR TITLE
Nereid user email should be case insensitive

### PR DESCRIPTION
**Pull request by prakashpp**

### CODE REVIEW

File: trytond_nereid/user.py [hunk 1]
```diff
@@ -498,7 +522,7 @@ def send_magic_login_link(cls, email):
 
         try:
             nereid_user, = cls.search([
-                ('email', 'ilike', email),
+                ('email', '=', email.lowe()),
```
Comment (tarunbhardwaj) [inline trytond_nereid/user.py]: typo

File: trytond_nereid/user.py [hunk 2]
```diff
@@ -269,6 +270,29 @@ def __setup__(cls):
...
         })
 
+    @classmethod
+    def validate(cls, records):
+        super(NereidUser, cls).validate(records)
+        cls.validate_email(records)
+
+    @classmethod
+    def validate_email(cls, records):
+        if backend.name() == 'sqlite':
```
Comment (tarunbhardwaj) [inline trytond_nereid/user.py]: I think `backend().name != 'postgresql'` makes more sense.

**Comment by tarunbhardwaj** [comment]
Just one suggestion and please fix flake8 issues.

File: trytond_nereid/user.py [hunk 3]
```diff
@@ -155,6 +155,15 @@ class NereidUser(ModelSQL, ModelView):
     active = fields.Boolean('Active')
 
     @classmethod
+    def __register__(cls, module_name):
+        super(NereidUser, cls).__register__(module_name)
+
+        # XXX: Remove in next version
+        Transaction().connection.cursor().execute(
+            'UPDATE %s SET email = LOWER(email)' % (cls._table, )
```
Comment (sharoonthomas) [inline trytond_nereid/user.py]: i know there is no sql injection here, but still use the params format

File: trytond_nereid/user.py [hunk 4]
```diff
@@ -262,8 +271,9 @@ def __setup__(cls):
         super(NereidUser, cls).__setup__()
         table = cls.__table__()
         cls._sql_constraints += [
-            ('unique_email_company', Unique(table, table.email, table.company),
-                'Email must be unique in a company'),
+            ('unique_email_company', Unique(
+                table, table.email, table.company, table.active
```
Comment (sharoonthomas) [inline trytond_nereid/user.py]: don't think you should consider active

Comment (prakashpp) [inline trytond_nereid/user.py]: @sharoonthomas in that case it wont allow migration to happen if case sensitive email exists

File: trytond_nereid/user.py (hunk 3 above)
Comment (prakashpp) [inline trytond_nereid/user.py]: 👍

**Pull request merged**